src/core/crypto/hdKey.ts

```diff
@@ -0,0 +1,105 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+import { hmac } from "@noble/hashes/hmac";
+import { sha512 } from "@noble/hashes/sha512";
+import * as bip39 from "@scure/bip39";
+
+export type DerivedKeys = {
+  key: Uint8Array;
+  chainCode: Uint8Array;
+};
+
+/**
+ * Aptos derive path is 637
+ */
+export const APTOS_HARDENED_REGEX = /^m\/44'\/637'\/[0-9]+'\/[0-9]+'\/[0-9]+'?$/;
+export const APTOS_BIP32_REGEX = /^m\/44'\/637'\/[0-9]+'\/[0-9]+\/[0-9]+$/;
+
+/**
+ * A list of supported key types and associated seeds
+ */
+export enum KeyType {
+  ED25519 = "ed25519 seed",
+}
+
+export const HARDENED_OFFSET = 0x80000000;
+
+/**
+ * Aptos derive path is 637
+ *
+ * Parse and validate a path that is compliant to BIP-32 in form m/44'/637'/{account_index}'/{change_index}/{address_index}
+ * for Secp256k1
+ *
+ * Note that for secp256k1, last two components must be non-hardened.
+ *
+ * @param path path string (e.g. `m/44'/637'/0'/0/0`).
+ */
+export function isValidBIP32Path(path: string): boolean {
+  if (!APTOS_BIP32_REGEX.test(path)) {
+    return false;
+  }
+  return true;
```

Comment:
Can reduce to `!!APTOS_BIP32_REGEX.test(path)` or maybe even just `APTOS_BIP32_REGEX.test(path)`